Scripts/Utilities/ComponentUtils.cs

```diff
@@ -33,9 +33,11 @@ public static class ComponentUtils
         public static T GetOrAddIf<T>(GameObject go, bool option) where T : Component
         {
             T component = go.GetComponent<T>();
+#if UNITY_EDITOR
             if (option && component == null)
                 component = Undo.AddComponent<T>(go);
-
+#endif
```

Comment:
Add the component without undo if in play mode?

Reply:
Ahh...Good find.